### ISSUE TITLE
Incorrect spans generated for top-level attributes

### ISSUE THREAD

**Issue by KyleMayes** [rust-lang/rust#36530]
@mikhail-m1 

Pull request #36354 broke the spans for top-level attributes.

Self-contained example (that used to work):

```rust
#![feature(rustc_private)]

extern crate syntax;

use syntax::parse::{self, ParseSess};

fn main() {
    let session = ParseSess::new();
    let mut parser = parse::new_parser_from_source_str(&session, vec![], "".into(), "#[test]".into());
    let attr = parser.parse_attribute(false).unwrap();
    assert_eq!(attr.span.lo.0, 0);
    assert_eq!(attr.span.hi.0, 7);
}
```

Now I get:

```
thread 'main' panicked at 'assertion failed: `(left == right)` (left: `6`, right: `7`)', test.rs:12
```

So the generated spans are one byte too short.

## Meta

```
rustc 1.13.0-nightly (6ffdda1ba 2016-09-14)
binary: rustc
commit-hash: 6ffdda1ba183c981d57e63b59c88184be449eee4
commit-date: 2016-09-14
host: x86_64-unknown-linux-gnu
release: 1.13.0-nightly
```

**Comment by TimNN** [comment]
You can also easily see the effects by looking at the highlighting reported by the error of 

```rust
#[foo]
mod foo {   
    #![foo]
}
```

**Comment by mikhail-m1** [comment]
Hi, I'll try fix it today or tomorrow

2016-09-16 13:20 GMT+03:00 Tim Neumann <notifications@github.com>:

**Comment by thepowersgang** [comment]
The test for this issue is failing in the nightly builds - https://buildbot.rust-lang.org/builders/nightly-dist-rustc-linux/builds/659/steps/distcheck/logs/stdio

**Comment by TimNN** [comment]
@thepowersgang: There's already a potential fix: #36678

**Comment by TimNN** [comment]
(By the way, I'm gonna close this, since the PR fixing this has been merged).
